### ISSUE TITLE
Cast Exception after collapsing the panel

### ISSUE THREAD

**Issue by 123a6bcw** [JetBrains-Research/IntelliJDeodorant#89]
To replicate the issue:
1) Find refactoring opportunities for God Class
2) Select one proposed refactoring (so the green selection will appear in the editor)
3) Collapse IntelliJDeodorant panel, and then try to open again.

```
java.lang.ClassCastException: class org.jetbrains.research.intellijdeodorant.ide.refactoring.extractClass.ExtractClassRefactoringType$AbstractExtractClassCandidateRefactoringGroup cannot be cast to class javax.swing.tree.TreeNode (org.jetbrains.research.intellijdeodorant.ide.refactoring.extractClass.ExtractClassRefactoringType$AbstractExtractClassCandidateRefactoringGroup is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @5c707f5b; javax.swing.tree.TreeNode is in module java.desktop of loader 'bootstrap')
	at java.desktop/javax.swing.tree.DefaultTreeModel.getIndexOfChild(DefaultTreeModel.java:166)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.getNodeForPath(VariableHeightLayoutCache.java:836)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.getRowForPath(VariableHeightLayoutCache.java:251)
	at java.desktop/javax.swing.tree.AbstractLayoutCache.getRowsForPaths(AbstractLayoutCache.java:461)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.resetRowSelection(DefaultTreeSelectionModel.java:777)
	at com.intellij.ui.treeStructure.treetable.TreeTable$ListToTreeSelectionModelWrapper.resetRowSelection(TreeTable.java:263)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1514)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1272)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:728)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.setModel(VariableHeightLayoutCache.java:111)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI.configureLayoutCache(BasicTreeUI.java:2103)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI.completeUIInstall(BasicTreeUI.java:812)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI.installUI(BasicTreeUI.java:772)
	at java.desktop/javax.swing.JComponent.setUI(JComponent.java:688)
	at java.desktop/javax.swing.JTree.setUI(JTree.java:706)
	at com.intellij.ui.treeStructure.Tree.setUI(Tree.java:110)
	at java.desktop/javax.swing.JTree.updateUI(JTree.java:727)
	at com.intellij.ui.treeStructure.treetable.TreeTableTree.updateUI(TreeTableTree.java:44)
	at com.intellij.ui.treeStructure.treetable.TreeTable.updateUI(TreeTable.java:171)
	at com.intellij.openapi.ui.ThreeComponentsSplitter.updateComponentTreeUI(ThreeComponentsSplitter.java:517)
	at com.intellij.openapi.ui.ThreeComponentsSplitter.doAddComponent(ThreeComponentsSplitter.java:540)
	at com.intellij.openapi.ui.ThreeComponentsSplitter.setLastComponent(ThreeComponentsSplitter.java:506)
	at com.intellij.openapi.wm.impl.ToolWindowsPane.setComponent(ToolWindowsPane.java:300)
	at com.intellij.openapi.wm.impl.ToolWindowsPane.addDecorator(ToolWindowsPane.java:202)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.doShowWindow(ToolWindowManagerImpl.kt:911)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.showToolWindowImpl(ToolWindowManagerImpl.kt:864)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.activateToolWindow(ToolWindowManagerImpl.kt:578)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.activateToolWindow$default(ToolWindowManagerImpl.kt:561)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.activated$intellij_platform_ide_impl(ToolWindowManagerImpl.kt:1769)
	at com.intellij.openapi.wm.impl.StripeButton$1.actionPerformed(StripeButton.java:62)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.JToggleButton$ToggleButtonModel.setPressed(JToggleButton.java:401)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:270)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6651)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at com.intellij.openapi.wm.impl.StripeButton.processMouseEvent(StripeButton.java:312)
	at java.desktop/java.awt.Component.processEvent(Component.java:6416)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5026)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:912)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:844)
	at com.intellij.ide.IdeEventQueue.lambda$null$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:739)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:502)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by onewhl** [comment]
Thanks. The issue can be solved by overriding `getIndexOfChild()` in `AbstractTreeTableModel`.

**Comment by onewhl** [comment]
Fixed in 2020.1-1.1
